src/opengl/gpu_tex.c

```diff
@@ -460,6 +460,56 @@ pl_tex gl_tex_create(pl_gpu gpu, const struct pl_tex_params *params)
     return NULL;
 }
 
+struct fmt_mapping {
+    const char *name;
+    GLenum format;
+    GLenum type;
+};
+
+// taken from OpenGL 4.6 Spec table 8.27
+static const struct fmt_mapping fmt_mappings[] = {
```

Comment:
I think this list is redundant with the existing list of `pl_fmt` inside `pl_gpu.formats`, no? You can access `PL_PRIV(fmt)` to get a `struct gl_format` containing the `fmt` and `type` of the corresponding `pl_fmt`.

Reply:
In `opengl/formats.c` only limited number of formats are registered. e.g. `rgb10_a2`, which is commonly used as swapchain frame, is not in the list.

We could add more formats to `pl_gpu`; however checks are necessary to ensure the added formats *actually* work on the gpu. I don't think it is feasible to write and maintain the code checking the availability of every format that the API user may pass to `pl_opengl_wrap`.

Instead, maintaining a full list that contains all possible formats here and omitting the availability checks is much simpler. And I don't think OpenGL will update this list in the future.